clang/include/clang/CIR/Interfaces/CIROpInterfaces.td

```diff
@@ -0,0 +1,103 @@
+      }]>,
+         InterfaceMethod<"", "bool", "hasLinkOnceLinkage", (ins), [{}],
+                         /*defaultImplementation=*/[{
+        return cir::isLinkOnceLinkage($_op.getLinkage());
+      }]>,
+         InterfaceMethod<"", "bool", "hasLinkOnceAnyLinkage", (ins), [{}],
+                         /*defaultImplementation=*/[{
+        return cir::isLinkOnceAnyLinkage($_op.getLinkage());
+      }]>,
+         InterfaceMethod<"", "bool", "hasLinkOnceODRLinkage", (ins), [{}],
+                         /*defaultImplementation=*/[{
+        return cir::isLinkOnceODRLinkage($_op.getLinkage());
+      }]>,
+         InterfaceMethod<"", "bool", "hasWeakLinkage", (ins), [{}],
+                         /*defaultImplementation=*/[{
+        return cir::isWeakLinkage($_op.getLinkage());
+      }]>,
+         InterfaceMethod<"", "bool", "hasWeakAnyLinkage", (ins), [{}],
+                         /*defaultImplementation=*/[{
+        return cir::isWeakAnyLinkage($_op.getLinkage());
+      }]>,
+         InterfaceMethod<"", "bool", "hasWeakODRLinkage", (ins), [{}],
+                         /*defaultImplementation=*/[{
+        return cir::isWeakODRLinkage($_op.getLinkage());
+      }]>,
+         InterfaceMethod<"", "bool", "hasInternalLinkage", (ins), [{}],
+                         /*defaultImplementation=*/[{
+        return cir::isInternalLinkage($_op.getLinkage());
+      }]>,
+         InterfaceMethod<"", "bool", "hasPrivateLinkage", (ins), [{}],
+                         /*defaultImplementation=*/[{
+        return cir::isPrivateLinkage($_op.getLinkage());
+      }]>,
+         InterfaceMethod<"", "bool", "hasLocalLinkage", (ins), [{}],
+                         /*defaultImplementation=*/[{
+        return cir::isLocalLinkage($_op.getLinkage());
+      }]>,
+         InterfaceMethod<"", "bool", "hasExternalWeakLinkage", (ins), [{}],
+                         /*defaultImplementation=*/[{
+        return cir::isExternalWeakLinkage($_op.getLinkage());
+      }]>,
+         InterfaceMethod<"", "bool", "hasCommonLinkage", (ins), [{}],
+                         /*defaultImplementation=*/[{
+        return cir::isCommonLinkage($_op.getLinkage());
+      }]>,
+         InterfaceMethod<"", "bool", "isDeclarationForLinker", (ins), [{}],
+                         /*defaultImplementation=*/[{
+        if ($_op.hasAvailableExternallyLinkage())
+          return true;
+        return $_op.isDeclaration();
+      }]>,
+         InterfaceMethod<"", "void", "setDSOLocal", (ins "bool":$val), [{}],
+                         /*defaultImplementation=*/[{
+        $_op.setDsolocal(val);
+      }]>,
+         InterfaceMethod<"", "bool", "isDSOLocal", (ins), [{}],
+                         /*defaultImplementation=*/[{
+        return $_op.getDsolocal();
+      }]>,
+         InterfaceMethod<"", "bool", "isWeakForLinker", (ins), [{}],
+                         /*defaultImplementation=*/[{
+        return cir::isWeakForLinker($_op.getLinkage());
+      }]>];
```

Comment:
Is this intended indentation?

Reply:
So to speak, yes. I reformatted the file with clang-format.

Reply:
Maybe that's why clang-format isn't commonly used on table-gen files.  That indentation is wrong and makes it (a tiny bit) harder to read.